workspace_from_url still broken with relative paths

I am still having problems with a simple `Resolver.workspace_from_url` (as used [here](https://github.com/OCR-D/ocrd_keraslm/blob/13133dbd36d660379f3267c1c185dbb52d4ff9a3/test/test_wrapper.py#L24)):

```
test/test_wrapper.py 09:58:08.817 DEBUG ocrd.resolver - workspace_from_url
mets_basename='mets.xml'
mets_url='test/assets/kant_aufklaerung_1784/data/mets.xml'
src_baseurl='test/assets/kant_aufklaerung_1784/data'
dst_dir='/tmp/pyocrd-test-ocrd_keraslm'
09:58:08.824 DEBUG ocrd.resolver.download_to_directory - directory=|/tmp/pyocrd-test-ocrd_keraslm| url=|test/assets/kant_aufklaerung_1784/data/mets.xml| basename=|mets.xml| if_exists=|raise| subdir=|None|
09:58:08.825 DEBUG ocrd.resolver.download_to_directory - Copying file '/home/xbert/unsortiert/arbeit/heyer/tools/ocrd_keraslm/test/assets/kant_aufklaerung_1784/data/mets.xml' to '/tmp/pyocrd-test-ocrd_keraslm/mets.xml'
09:58:08.863 DEBUG ocrd.resolver.download_to_directory - directory=|/tmp/pyocrd-test-ocrd_keraslm| url=|OCR-D-IMG/OCR-D-IMG_0001.tif| basename=|OCR-D-IMG_0001.tif| if_exists=|skip| subdir=|OCR-D-IMG|
09:58:08.864 DEBUG ocrd.resolver.download_to_directory - directory=|/tmp/pyocrd-test-ocrd_keraslm| url=|test/assets/kant_aufklaerung_1784/data/OCR-D-IMG/OCR-D-IMG_0001.tif| basename=|OCR-D-IMG_0001.tif| if_exists=|skip| subdir=|OCR-D-IMG|
...
E               FileNotFoundError: File path passed as 'url' to download_to_directory does not exist: OCR-D-IMG/OCR-D-IMG_0001.tif
E               FileNotFoundError: File path passed as 'url' to download_to_directory does not exist: test/assets/kant_aufklaerung_1784/data/OCR-D-IMG/OCR-D-IMG_0001.tif
E                   Exception: Already tried prepending baseurl 'test/assets/kant_aufklaerung_1784/data'. Cannot retrieve 'test/assets/kant_aufklaerung_1784/data/OCR-D-IMG/OCR-D-IMG_0001.tif'
```

So, compared to #266, looking at the solution in #303 (i.e. keeping `workspace_from_url` more or less as it is, based on `download_file` not `download_url`, but incorporating some baseurl + recursion logic into `download_file`), it seems to me the new logic is still flawed: It is not enough to prepend `baseurl` like this, because baseurl itself is relative to the old CWD where we resolved the source METS, not to `dst_dir`.

--- a/ocrd/ocrd/resolver.py
+++ b/ocrd/ocrd/resolver.py
@@ -122,6 +122,10 @@ class Resolver():
         if mets_url is None:
             raise ValueError("Must pass 'mets_url' workspace_from_url")
 
+        # if mets_url is a relative filename, make it absolute
+        if is_local_filename(mets_url) and not Path(mets_url).is_absolute():
+            mets_url = str(Path(Path.cwd() / mets_url))
+
         # if mets_basename is not given, use the last URL segment of the mets_url
         if mets_basename is None:
             mets_basename = nth_url_segment(mets_url, -1)

--- a/tests/cli/test_workspace.py
+++ b/tests/cli/test_workspace.py
@@ -191,6 +191,14 @@ class TestCli(TestCase):
             self.assertEqual(len(workspace.mets.find_files()), 33)
             self.assertFalse(exists(file_path))
 
+    def test_clone_relative(self):
+        # Create a relative path to trigger make sure #319 is gone
+        src_path = str(Path(assets.path_to('kant_aufklaerung_1784/data/mets.xml')).relative_to(Path.cwd()))
+        with TemporaryDirectory() as tempdir:
+            result = self.runner.invoke(workspace_cli, ['clone', '-a', src_path, tempdir])
+            self.assertEqual(result.exit_code, 0)
+            self.assertTrue(exists(join(tempdir, 'OCR-D-GT-PAGE/PAGE_0017_PAGE.xml')))
+
     def test_copy_vs_clone(self):
         src_dir = assets.path_to('kant_aufklaerung_1784/data')
         with TemporaryDirectory() as tempdir:

--- a/tests/test_workspace.py
+++ b/tests/test_workspace.py
@@ -101,6 +101,22 @@ class TestWorkspace(TestCase):
             self.assertEqual(f, Path('TEMP', '%s.tif' % SAMPLE_FILE_ID))
             self.assertTrue(Path(ws1.directory, f).exists())
 
+    def test_from_url_dst_dir_download(self):
+        """
+        https://github.com/OCR-D/core/issues/319
+        """
+        with TemporaryDirectory() as tempdir:
+            # TODO re-enable once #393 is merged
+            #  ws_dir = join(tempdir, 'non-existing-for-good-measure')
+            ws_dir = tempdir
+            # Create a relative path to trigger #319
+            src_path = str(Path(assets.path_to('kant_aufklaerung_1784/data/mets.xml')).relative_to(Path.cwd()))
+            self.resolver.workspace_from_url(src_path, dst_dir=ws_dir, download=True)
+            #  from os import system
+            #  system('find %s' % ws_dir)
+            self.assertTrue(Path(ws_dir, 'mets.xml').exists())  # sanity check, mets.xml must exist
+            self.assertTrue(Path(ws_dir, 'OCR-D-GT-PAGE/PAGE_0017_PAGE.xml').exists())
+
     def test_227_1(self):
         def find_recursive(root):
             ret = []

